web/src/containers/devices/utils.js

```diff
@@ -351,3 +365,52 @@ export const getResourceRegistrationNotificationKey = deviceId =>
 // Redux and event key for the notification state for an update of a single resource
 export const getResourceUpdateNotificationKey = (deviceId, href) =>
   `${DEVICES_RESOURCE_UPDATE_WS_KEY}.${deviceId}.${href}`
+
+export const isValidEndpoint = endpoint =>
+  !!!endpoint.match(/[^a-zA-Z0-9\-+:./]/)
+
+// Updates the resource through rest API
+export const updateResourceMethod = async (
+  { deviceId, href, currentInterface = '', ttl = RESOURCE_DEFAULT_TTL_RAW },
+  resourceDataUpdate,
+  successCallback,
+  errorCallback
+) => {
+  try {
+    await updateDevicesResourceApi(
+      { deviceId, href, currentInterface, ttl },
+      resourceDataUpdate
+    )
+
+    if (isMounted.current && isFunction(successCallback)) {
+      successCallback()
+    }
+  } catch (error) {
+    if (error && isMounted.current) {
+      isFunction(errorCallback) && errorCallback(error)
+    }
+  }
+}
+
+export const getColorByProvisionStatus = provisionStatus => {
+  if (provisionStatus === devicesProvisionStatuses.UNINITIALIZED) {
+    return 'grey'
```

Comment:
pls reference to constant if it possible - same for green, orange, red,

it is better you the `map[provisionStatus]color` and if `provisionStatus` not match what happens ? - I suggest to set color grey.